students/Wieslaw_Pucilowski/lesson09/mailroom5_OO.py

```diff
@@ -191,17 +157,94 @@ def write_letters(self):
             d.write_letter()
 
 
+# Donors DB initialization
+dd = DonorsCollection()
+
+
+class Main:
+    @staticmethod
+    def report():
+        dd.print_report()
+
+    @staticmethod
+    def letters():
+        dd.write_letters()
+
+    @staticmethod
+    def donor():
+        dd.add_donor()
+
+    @staticmethod
+    def show():
+        dd.display()
+
+    def quit(msg):
+        print("{}".format(msg))
+        return "exit menu"
+
+    @classmethod
+    def main_quit(cls):
+        return cls.quit("Goodbye...")
+
+    @classmethod
+    def sub_quit(cls):
+        return cls.quit("Back to Main menu...")
+
+    def menu_selection(prompt, dispatcher):
+            while True:
+                response = input(prompt)
+                try:
+                    if dispatcher[response]() == "exit menu":
+                        break
+                except KeyError:
+                    print(response, "Wrong response !")
+
+    @classmethod
+    def sub_menu(cls):
+        cls.menu_selection(submenu, subfeatures)
```

Comment:
this and sub_quit should not be classmethods, all of these functions should be staticmethods (which was another reason I suggested regular methods otherwise you're just wrapping all of the staticmethods into a single class, like grouping which is Ok but not really necessary).

I recommend reviewing the classmethod vs staticmethod:
https://www.geeksforgeeks.org/class-method-vs-static-method-python/